torch/_dynamo/variables/tensor.py

```diff
@@ -744,74 +764,63 @@ def call_method(
+                isinstance(arg, TensorWithTFOverrideVariable) for arg in all_args
+            )
+        else:
+            return False
 
-           x1 = torch.sigmoid(x0)
+    @staticmethod
+    def dispatch_torch_function(tx, fn, args, kwargs):
+        """Gathers all args that are TensorWithTFOverrideVariable and dispatches based on the ordering in _get_overloaded_args"""
 
-        And `x0` has an override, then:
-        * `original_func_var` will be a `VariableTracker` object wrapping `torch.sigmoid`
-        * `tensor_with_tf_override_source` will be the `Source` object from
-          the original tensor override instance in the beginning of the program
-        * `tf_func` will be the custom `__torch_function__` function
-        * `subclass_type` will be `type(x0)`
+        all_args = args + tree_flatten(kwargs)[0]
+        overloaded_args = _get_overloaded_args(
+            [arg for arg in all_args if isinstance(arg, TensorWithTFOverrideVariable)],
+            lambda x: x.subclass_type,
+        )
 
-        The caller is expected to properly massage args and kwargs before
-        passing them into this function.
+        for arg in overloaded_args:
+            res = arg.call_torch_function(
+                tx,
+                fn,
+                variables.TupleVariable(
+                    list({arg.subclass_type_var() for arg in overloaded_args})
+                ),
+                args,
+                kwargs,
+            )
 
-        The caller is responsible for wrapping the return value, if needed.
-        """
-        from . import UserDefinedClassVariable
-        from .builder import TupleVariable, VariableBuilder
+            if not (
+                isinstance(res, variables.ConstantVariable)
+                and res.value is NotImplemented
+            ):
+                return res
 
-        source = AttrSource(
-            AttrSource(tensor_with_tf_override_source, "__torch_function__"),
-            "__func__",
+        unimplemented(
+            f"All __torch_function_overrides__ for call {fn} with args {args} and kwargs {kwargs} returned NotImplemented"
         )
-        tf_func_var = VariableBuilder(tx, source)(tf_func)
-        type_var = UserDefinedClassVariable(subclass_type, **options)
 
+    def call_torch_function(self, tx, fn, types, args, kwargs):
         # signature:
         # def __torch_function__(cls, func, types, args=(), kwargs=None):
         tf_args = (
-            type_var,  # cls
-            original_func_var,  # func
-            (type_var,),  # types
-            TupleVariable(args),  # args
-            kwargs,  # kwargs
+            self.subclass_type_var(),  # cls
+            fn,  # func
+            types,
+            variables.TupleVariable(list(args)),
         )
-
-        # Disable __torch_function__ here to prevent the clone of the
-        # example tensor from going into the override.
         with torch._C.DisableTorchFunctionSubclass():
```

Comment:
This is preexisting code, but this is a pretty strange thing to do. You're not actually going to call anything on the real tensor subclass. Why do you need this?

Reply:
I can try without this, but don't we call clone on it? (or we may not do this anymore, I think we used to)

Reply:
We have to fakeify it, you didn't really touch that code here but that's another part of the holistic design which also needs to be handled properly.